src/cdk/private/style-loader.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  ApplicationRef,
+  ComponentRef,
+  createComponent,
+  EnvironmentInjector,
+  inject,
+  Injectable,
+  Type,
+} from '@angular/core';
+
+/** Apps in which we've loaded styles. */
+const appsWithLoaders = new WeakMap<
+  ApplicationRef,
+  {
+    /** Style loaders that have been added. */
+    loaders: Set<Type<unknown>>;
+
+    /** References to the instantiated loaders. */
+    refs: ComponentRef<unknown>[];
+  }
+>();
+
+/**
+ * Service that loads structural styles dynamically
+ * and ensures that they're only loaded once per app.
+ */
+@Injectable({providedIn: 'root'})
+export class _CdkPrivateStyleLoader {
+  private _appRef = inject(ApplicationRef);
+  private _environmentInjector = inject(EnvironmentInjector);
+
+  /**
+   * Loads a set of styles.
+   * @param loader Component which will be instantiated to load the styles.
+   */
+  loadStyles(loader: Type<unknown>): void {
```

Comment:
Usage looks a little redundant with `styleLoader.loadStyles()`

What about renaming this to just `load`